### ISSUE TITLE
IPC error when attaching multiple drivers

### ISSUE THREAD

**Issue by robertnishihara** [amplab/ray#259]
This bug arose while I was testing `cluster.py` and restarting the workers multiple times.

To reproduce this error. In `build`, start a scheduler
```
./scheduler 127.0.0.1:10001
```
In `build`, start an object store
```
./objstore 127.0.0.1:10001 127.0.0.1:20001
```
Start a Python shell and do
```python
>>> import ray
>>> ray.connect("127.0.0.1:10001", "127.0.0.1:20001", "127.0.0.1:30001", is_driver=True)
>>> exit()
```
Then start another Python shell and do
```python
>>> import ray
>>> ray.connect("127.0.0.1:10001", "127.0.0.1:20001", "127.0.0.1:30002", is_driver=True)
```
The only difference is that I incremented the shell port number to avoid reusing the same address. This fails with
```
fatal error occured: Check failed at line 71 in /Users/rkn/Workspace/ray/src/ipc.cc: false with message boost::interprocess exception: No such file or directory
```

Note that if we omit the `exit()` command the first time, then this error does not happen.

**Comment by robertnishihara** [comment]
Within the second call to `ray.connect` (in `worker.py`) the error occurs in the line
```
worker.handle = ray.lib.create_worker(worker.scheduler_address, worker.objstore_address, worker.worker_address, is_driver)
```
Within `create_worker` in `raylib.cc`, it happens in the line
```
worker->register_worker(std::string(worker_addr), std::string(objstore_addr), is_driver);
```
Within `register_worker` in `worker.cc`, the error occurs in the line
```
RAY_CHECK(request_obj_queue_.connect(std::string("queue:") + objstore_address + std::string(":obj"), false), "error connecting request_obj_queue_");
```
Within `connect` in `ipc.cc`, the error occurs in the line
```
queue_ = std::unique_ptr<bip::message_queue>(new bip::message_queue(bip::open_only, name_.c_str()));
```

**Comment by robertnishihara** [comment]
The problem goes away when we comment out the `message_queue::remove` from the destructor like this.
```
MessageQueue<>::~MessageQueue() {
  if (!name_.empty()) {
    // bip::message_queue::remove(name_.c_str());
  }
}
```

**Comment by robertnishihara** [comment]
Addressed by #260.
